src/Components/QuickSelect.tsx

```diff
@@ -11,166 +12,94 @@ import {
   Menu,
   MenuItem,
 } from "@material-ui/core";
-
-class DateRange {
-  text: string;
-  startDate: Date;
-  endDate: Date;
-
-  constructor(text) {
-    this.text = text;
-    if (this.text.includes("next")) {
-      this.startDate = new Date();
-      this.endDate = new Date();
-      this.endDate.setTime(this.startDate.getTime() + ms(this.text.slice(5)));
-    } else {
-      this.startDate = new Date();
-      this.endDate = new Date();
-      this.startDate.setTime(this.startDate.getTime() - ms(this.text.slice(5)));
-    }
-  }
-}
+import { DateSelect } from "./DateSelect2.tsx";
+import DateRange from "./DateRange.tsx";
 
 interface Inputs {
   boxClass: string;
-  recentlySelected: DateRange[];
-  quickSelectText: string[];
+  recentlySelected: string[][];
   termAnchorEl: any;
   intervalAnchorEl: any;
+  timeIntervalText: string[][];
+  timeFormat: string;
+  quickSelectContent: string[];
+  setDateRange(h): void;
+  setQuickSelectContent(r): void;
+  dateRange: DateRange;
+  applyChanges(): void;
+  getData(f): void;
+  setRecentlySelected(i): void;
   setTermAnchorEl(w): void;
   setIntervalAnchorEl(w): void;
-  setQuickSelectText(z): void;
-  setRecentlySelected(x): void;
-  setDates(y): void;
 }
 
 export function QuickSelect(props: Inputs) {
-  const timeIntervalObjects = [
-    new DateRange("Last 15 Minutes"),
-    new DateRange("Last 30 Minutes"),
-    new DateRange("Last 1 Hour"),
-    new DateRange("Last 24 hours"),
-    new DateRange("Last 7 days"),
-    new DateRange("Last 30 days"),
-    new DateRange("Last 90 days"),
-    new DateRange("Last 1 year"),
+  const terms = ["Last", "Next"];
+  const intervals = [

```

Comment:
I see the intervals present in a lot of places - good opportunity to pull them out into a prop and something that's shared amongst all of the files.